diffus-derive-test/src/lib.rs

```diff
@@ -11,6 +11,65 @@ mod test {
         Diffable,
     };
 
+    #[derive(Diffus, Debug, PartialEq)]
+    struct Pair {
+        id: u32,
+        value: u32,
+    }
+
+    impl diffus::Same for Pair {
+        fn same(&self, other: &Self) -> bool {
+            self.id == other.id
+        }
+    }
+
+    #[test]
+    fn non_trivial_same_collection() {
+        let left = vec![
+            Pair { id: 1, value: 0 },
+            Pair { id: 2, value: 0 },
+            Pair { id: 3, value: 0 },
+            Pair { id: 4, value: 0 },
+            Pair { id: 5, value: 0 },
+            Pair { id: 6, value: 0 },
+            Pair { id: 7, value: 0 },
+        ];
+        let right = vec![
+            Pair { id: 1, value: 0 },
+            Pair { id: 2, value: 1 },
+            Pair { id: 4, value: 0 },
+            Pair { id: 3, value: 0 },
+            Pair { id: 5, value: 0 },
+            Pair { id: 6, value: 0 },
+        ];
+
+        let diff = left.diff(&right);
+
+        use diffus::edit::collection;
+        use diffus::edit::Edit;
```

Comment:
`Edit` is only used once below, but `collection::Edit` is used many times, do a switch-a-roo?

Reply:
good point, went with something in between
collection::Edit and edit::Edit to signify the difference